lib/komponent/core/component_path_resolver.rb

```diff
@@ -0,0 +1,66 @@
+module Komponent
+  class ComponentPathResolver
+    attr_accessor :path
+
+    def initialize(component)
+      @component = component
+      resolve
+    end
+
+    protected
+
+    def resolve
+      component_found = nil
+
+      component_paths.each do |component_path|
+        component_found = find_component(component_path)
+        break if component_found
+      end
+
+      raise ComponentPathResolver::MissingComponentError.new(component_paths) unless component_found
+
+      component_parts.each do |part|
+        component_found += part
+      end
+
+      @path = Pathname.new(component_found)
+    end
+
+    def find_component(component_path)
+      pattern = /#{component_parts.join("_")}_component.rb/
+      Find.find(component_path.to_s) do |path|
```

Comment:
We don't need to scan all the files since we know the exact name the component file should be in the given path. We just need to check whether this file exists.